scraper.ts

```diff
@@ -209,10 +216,29 @@ export const scraper = async (
 
 
 
+}
+
+export const scrapeLink = async (link: string, depthCounter: number) => {
+    debugger;
+
+
+    const fetched = await fetchHtml(link)
+    writeLog('fetched page ' + link)
+    const cleanedHtml = await removeTag(fetched, blackListElems)
+    writeLog('cleaned page ' + link)
+    saveHtml(cleanedHtml, `${link.match(/\/\/(.*?)\.com/)?.[1] ?? ''}`)
+    writeLog('saved page ' + link)
+
+    const linkArr = await extractLinks(cleanedHtml, 10)
+    linkArr.forEach((link) => {
 
+        if (depthCounter > 0) return scrapeLink(link, depthCounter--)
```

Comment:
I would expect:

if (depthCounter > 0) return scrapeLink(link, depthCounter-1)

Using a decrementer might actually cause problems, because you don't want to decrement the value that is stored in the parent function, you only want it to be lower by the time it gets to the child functions.